src/controller/java/src/chip/devicecontroller/ChipTLVValueDecoder.java

```diff
@@ -0,0 +1,28 @@
+/*
+ *   Copyright (c) 2020-2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package chip.devicecontroller;
+
+import chip.devicecontroller.model.ChipAttributePath;
+import chip.devicecontroller.model.ChipEventPath;
+
+public class ChipTLVValueDecoder {
+  public static native <T> T decodeAttributeValue(ChipAttributePath attributePath, byte[] tlv);
+
+  public static native <T> T decodeEventValue(ChipEventPath eventPath, byte[] tlv);
```

Comment:
@yufengwangca 
I already know about that plan.
https://github.com/project-chip/connectedhomeip/issues/27398
However, as explained in the above issue, the current ChipTool needs encode and decode to use instead of using the existing ChipClusters.java file, and the tlv library is insufficient.

So, in my opinion, after adding the api for encoding for struct, I want it to proceed in the direction of adding a separate feature.
(Disable this feature when not in use.)

Reply:
I understand what you mean.
After making kotlin version, it seems that this API can be changed to call kotlin library instead of JNI.